shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/domain/usecase/LogOutUseCase.kt

```diff
@@ -1,18 +1,28 @@
 package vn.luongvo.kmm.survey.domain.usecase
 
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.*
 import vn.luongvo.kmm.survey.domain.repository.AuthRepository
+import vn.luongvo.kmm.survey.domain.repository.UserRepository
 
 interface LogOutUseCase {
 
     operator fun invoke(): Flow<Unit>
 }
 
-class LogOutUseCaseImpl(private val repository: AuthRepository) : LogOutUseCase {
+class LogOutUseCaseImpl(
+    private val authRepository: AuthRepository,
+    private val userRepository: UserRepository
+) : LogOutUseCase {
 
     override operator fun invoke(): Flow<Unit> {
-        return repository.logOut()
-            .onEach { repository.clearToken() }
+        return authRepository.logOut()
+            .catch {
+                // On error resume next
+                emit(Unit)
```

Comment:
This emit will bypass `catch {}` and call `onEach {}` again, we don't need to duplicate clearing tokens twice 👏 Good job 💯